app.go

```diff
@@ -262,15 +260,27 @@ func (ah *apiHandlers) dropHandler(w http.ResponseWriter, r *http.Request) {
 	vars := mux.Vars(r)
 	coll, err := ah.getCollection(vars["collection"])
 	if err != nil {
+
 		http.Error(w, err.Error(), http.StatusBadRequest)
 		return
 	}
-	ah.engine.Drop(coll)
+
+	ok, err := ah.engine.Drop(coll)
+
+	if err !=nil {
```

Comment:
How can we be sure it's a bad request here?  if elastic search was unavailable, we'd get an error, but it's not a bad request.